node/subsystem/src/lib.rs

```diff
@@ -261,3 +261,76 @@ where
 		}
 	}
 }
+
+// Jaeger integration.
+//
+// See <https://www.jaegertracing.io/> for an introduction.
+//
+// The easiest way to try Jaeger is:
+//
+// - Start a docker container with the all-in-one docker image (see below).
+// - Open your browser and navigate to <https://localhost:16686> to acces the UI.
+//
+// The all-in-one docker image can be started with:
+//
+// ```not_rust
+// docker run -d --name jaeger \
+//  -e COLLECTOR_ZIPKIN_HTTP_PORT=9411 \
+//  -p 5775:5775/udp \
+//  -p 6831:6831/udp \
+//  -p 6832:6832/udp \
+//  -p 5778:5778 \
+//  -p 16686:16686 \
+//  -p 14268:14268 \
+//  -p 14250:14250 \
+//  -p 9411:9411 \
+//  jaegertracing/all-in-one:1.21
+// ```
+//
+
+/// Shortcut for [`hash_span`] with the hash of the `PoV`.
+pub fn pov_span(pov: &PoV, span_name: &str) -> mick_jaeger::Span {
+	hash_span(&pov.hash(), span_name)
+}
+
+/// Creates a `Span` referring to the given hash. All spans created with [`hash_span`] with the
+/// same hash (even from multiple different nodes) will be visible in the same view on Jaeger.
+pub fn hash_span(hash: &Hash, span_name: &str) -> mick_jaeger::Span {
+	let trace_id = {
+		let mut buf = [0u8; 16];
+		buf.copy_from_slice(&hash.as_ref()[0..16]);
+		std::num::NonZeroU128::new(u128::from_be_bytes(buf)).unwrap()
+	};
+
+	TRACES_IN.lock().unwrap().as_ref().unwrap().span(trace_id, span_name)
+}
+
+/// Must be called otherwise the other jaeger-related functions will panic.
+///
+/// THIS API IS REALLY BAD AND SHOULDN'T BE FOUND IN THE MASTER BRANCH
```

Comment:
what do you think the way to move forward would be?

can we use these libs?
https://github.com/tokio-rs/tracing/tree/HEAD/tracing-opentelemetry
https://github.com/open-telemetry/opentelemetry-rust/tree/HEAD/opentelemetry-jaeger

Reply:
> what do you think the way to move forward would be?

Properly integrate it in Substrate, and pass an `Arc<TracesIn>` as parameter to the various subsystems.

> can we use these libs?

I'm going to paste this conversation here, for posterity:

![image](https://user-images.githubusercontent.com/1412254/100612022-90668d80-3312-11eb-9e4e-79cf3320526c.png)